cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ServiceUtils.java

```diff
@@ -0,0 +1,244 @@
+            List<ServicePort> ports
+    )   {
+        return createService(
+                name,
+                namespace,
+                labels,
+                ownerReference,
+                template,
+                ports,
+                labels.strimziSelectorLabels(),
+                "ClusterIP",
+                null,
+                null,
+                ipFamilyPolicy(template),
+                ipFamilies(template)
+        );
+    }
+
+    /**
+     * Creates a discoverable ClusterIP Service. discoverable service has some additional labels and annotations to
+     * simplify discovery.
+     *
+     * @param name                  Name of the Service
+     * @param namespace             Namespace of the Service
+     * @param labels                Labels of the Service
+     * @param ownerReference        OwnerReference of the Service
+     * @param template              Service template with user's custom configuration
+     * @param ports                 List of service ports
+     * @param discoveryLabels       Additional discovery labels
+     * @param discoveryAnnotations  Additional discovery annotations
+     *
+     * @return  New discoverable ClusterIP Service
+     */
+    public static Service createDiscoverableClusterIpService(
+            String name,
+            String namespace,
+            Labels labels,
+            OwnerReference ownerReference,
+            InternalServiceTemplate template,
+            List<ServicePort> ports,
+            Map<String, String> discoveryLabels,
+            Map<String, String> discoveryAnnotations
+    )   {
+        return createService(
+                name,
+                namespace,
+                labels.withStrimziDiscovery(),
+                ownerReference,
+                template,
+                ports,
+                labels.strimziSelectorLabels(),
+                "ClusterIP",
+                discoveryLabels,
+                discoveryAnnotations,
+                ipFamilyPolicy(template),
+                ipFamilies(template)
+        );
+    }
+
+    /**
+     * Creates a service
+     *
+     * @param name                  Name of the Service
+     * @param namespace             Namespace of the Service
+     * @param labels                Labels of the Service
+     * @param ownerReference        OwnerReference of the Service
+     * @param template              Template with user's custom metadata for this service
+     * @param ports                 List of service ports
+     * @param selector              Selector for selecting the Pods to route the traffic to
+     * @param type                  Type of the service
+     * @param additionalLabels      Additional labels
+     * @param additionalAnnotations Additional annotations
+     * @param ipFamilyPolicy        IP Family Policy configuration
+     * @param ipFamilies            List of IP familiers
+     *
+     * @return  New discoverable ClusterIP Service
```

Comment:
I guess it's enough `New ClusterIP Service` because this is the common method across the previous two. Only with `createDiscoverableClusterIpService` we get  a discoverable one.